docker/Dockerfile

```diff
@@ -93,7 +110,7 @@ RUN \
     openssl \
     && rm -rf /var/lib/apt/lists/*
 
-COPY --from=builder /install /usr/local
+COPY --from=builder /synapse/ /synapse
```

Comment:
I hadn't considered that people were building on top of our docker images. A [quick search](https://github.com/search?l=Dockerfile&q=FROM+matrixdotorg%2Fsynapse&type=Code) fins a few.

I couldn't find an equivalent to `pip install --prefix` in poetry last time I looked. But this comment https://github.com/python-poetry/poetry/issues/234#issuecomment-571964690 might be useful.

Reply:
If we could dump everything back in `/usr/local/` then we wouldn't need https://github.com/matrix-org/complement/pull/337.

Reply:
> I hadn't considered that people were building on top of our docker images. A [quick search](https://github.com/search?l=Dockerfile&q=FROM+matrixdotorg%2Fsynapse&type=Code) fins a few.

and I'll bet there are a whole bunch more that aren't on (public) github.

Copying a virtualenv wholesale into /usr/local doesn't sound like a thing that will work to me. If nothing else there will be a bunch of hardcoded paths which would need updating.

Ultimately, I think our best approach here is just going to be to warn people loudly that this is a change we are making. Which sounds to me like we should also get our story straight about how people are meant to install extensions (https://github.com/matrix-org/synapse/issues/9944) to avoid breaking things twice for people.

To decouple this... could we do the same thing as the debian build, and hack around with `poetry export` and `pip install -r` ?